XCreateFile fail to open config.txt

For some unknown reason XCreateFile (inside load_conf_file function) is unable to open the config.txt.

I remember this working the last time I checked (first days of Jan). I'll give it a look soon.

My hunch is that it will not work on Cxbx-Reloaded. Once an XBE starts, the filesystem seen by the XBE is the virtual filesystem created by Cxbx, not the host file system. Unless the user has their test suite XBE stored in the internal filesystem, I don't see how the path to look for the config file inside of the test suite would match.

Also, when building with the latest commit, linking fails for me. I also see warnings during compile, which did not exist before.

lld-link: error: /mnt/e/Windows/Programming/nxdk/my_progs/xbox_kernel_test_suite/main.obj: undefined symbol: _strtok_r
lld-link: error: /mnt/e/Windows/Programming/nxdk/my_progs/xbox_kernel_test_suite/main.obj: undefined symbol: _strtok
lld-link: error: /mnt/e/Windows/Programming/nxdk/my_progs/xbox_kernel_test_suite/main.obj: undefined symbol: _strtol
/mnt/e/Windows/Programming/nxdk/my_progs/xbox_kernel_test_suite/../../Makefile:94: recipe for target 'main.exe' failed

I went ahead and fixed all of the depreciation issues with the test suite tonight but now I'm running into nxdk compile errors on the newest nxdk commit. One of them for example:

/mnt/e/Windows/Programming/nxdk2/my_progs/xbox_kernel_test_suite/../../lib/winapi/fileio.c:197:13: error: expected expression
            FILE_NETWORK_OPEN_INFORMATION networkInfo;
/mnt/e/Windows/Programming/nxdk2/my_progs/xbox_kernel_test_suite/../../lib/winapi/fileio.c:259:13: error: expected expression
            FILE_NETWORK_OPEN_INFORMATION networkInfo;

Moving the variable declaration outside of the case statement fixes those errors. Then I get more warnings and then these errors:

/mnt/e/Windows/Programming/nxdk/my_progs/xbox_kernel_test_suite/../../lib/libcxx/include/type_traits:737:50: error: use of undeclared identifier 'char16_t'
template <>          struct __libcpp_is_integral<char16_t>           : public true_type {};
                                                 ^
/mnt/e/Windows/Programming/nxdk/my_progs/xbox_kernel_test_suite/../../lib/libcxx/include/type_traits:738:50: error: use of undeclared identifier 'char32_t'
template <>          struct __libcpp_is_integral<char32_t>           : public true_type {};

Are there bugs in the winapi? This code was committed in July it looks like and I'm wondering why I'm hitting these errors in general files where others should be hitting the same errors.

--- a/main.c
+++ b/main.c
@@ -108,8 +108,8 @@ void main(void){
     pb_show_debug_screen();
 
     vector_init(&tests_to_run);
-    load_conf_file("config.txt");
-    open_output_file("kernel_tests.log");
+    load_conf_file("D:\\config.txt");
+    open_output_file("D:\\kernel_tests.log");
 
     print("Kernel Test Suite");
     run_tests();

--- a/main.c
+++ b/main.c
@@ -12,10 +12,10 @@
 #include "vector.h"
 #include "string_extra.h"
 
-int load_conf_file(char *config_file_path) {
-    print("Trying to open config file: %s", config_file_path);
+int load_conf_file(char *file_path) {
+    print("Trying to open config file: %s", file_path);
     HANDLE handle = CreateFile(
-        config_file_path,
+        file_path,
         GENERIC_READ,
         FILE_SHARE_READ,
         0,
@@ -24,13 +24,13 @@ int load_conf_file(char *config_file_path) {
         NULL
     );
     if(handle == INVALID_HANDLE_VALUE) {
-        print("Could not open config file '%s' for read", config_file_path);
+        print("Could not open config file '%s' for read", file_path);
         return -1;
     }
 
     DWORD file_size = GetFileSize(handle, NULL);
     if(file_size == INVALID_FILE_SIZE) {
-        print("ERROR: Could not get file size for %s", config_file_path);
+        print("ERROR: Could not get file size for %s", file_path);
         return -1;
     }
 

--- a/output.c
+++ b/output.c
@@ -49,15 +49,15 @@ void print_test_footer(
     }
 }
 
-void open_output_file(char* file_name) {
+void open_output_file(char* output_file_path) {
     if(is_emu) {
-        print("Kernel Test Suite: Skipping creating %s because on emulator", file_name);
+        print("Kernel Test Suite: Skipping creating %s because on emulator", output_file_path);
         return;
     }
 
-    debugPrint("Creating file %s", file_name);
+    debugPrint("Creating file %s", output_file_path);
     output_filehandle = CreateFile(
-        file_name,
+        output_file_path,
         GENERIC_WRITE,
         FILE_SHARE_WRITE,
         0,
@@ -67,7 +67,7 @@ void open_output_file(char* file_name) {
     );
 
     if(output_filehandle == INVALID_HANDLE_VALUE) {
-        debugPrint("ERROR: Could not create file %s", file_name);
+        debugPrint("ERROR: Could not create file %s", output_file_path);
     }
 }
 

--- a/output.c
+++ b/output.c
@@ -49,15 +49,15 @@ void print_test_footer(
     }
 }
 
-void open_output_file(char* output_file_path) {
+void open_output_file(char* file_path) {
     if(is_emu) {
-        print("Kernel Test Suite: Skipping creating %s because on emulator", output_file_path);
+        print("Kernel Test Suite: Skipping creating %s because on emulator", file_path);
         return;
     }
 
-    debugPrint("Creating file %s", output_file_path);
+    debugPrint("Creating file %s", file_path);
     output_filehandle = CreateFile(
-        output_file_path,
+        file_path,
         GENERIC_WRITE,
         FILE_SHARE_WRITE,
         0,
@@ -67,7 +67,7 @@ void open_output_file(char* output_file_path) {
     );
 
     if(output_filehandle == INVALID_HANDLE_VALUE) {
-        debugPrint("ERROR: Could not create file %s", output_file_path);
+        debugPrint("ERROR: Could not create file %s", file_path);
     }
 }
 

